.github/scripts/therock_matrix.py

```diff
@@ -32,5 +32,9 @@
     "all": {
         "cmake_options": "-DTHEROCK_ENABLE_CORE=ON -DTHEROCK_ENABLE_PROFILER=ON -DTHEROCK_ENABLE_ALL=OFF",
         "project_to_test": "hip-tests, rocprofiler-tests",
+    },
+    "full-build": {
+        "cmake_options": "-DTHEROCK_ENABLE_ALL=ON",
+        "project_to_test": "hip-tests, rocprofiler-tests",
```

Comment:
you can re-use `all`! currently, the `subtree_to_project_map` doesn't do anything, we only pull `all` for builds / tests (if you review the code in `therock_configure_ci.py`)

```suggestion
    "all": {
        "cmake_options": "-DTHEROCK_ENABLE_ALL=ON",
        "project_to_test": "hip-tests, rocprofiler-tests",
    },
```